### PR TITLE
[notification] open-telemetry links

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

* Visualise the APM links on a GitHub comment and Slack comments
* Fix some NPE when the BlueOcean entrypoint returns empty values.

## Why is it important?

Provide enough context regarding the APM traces and further insights for the build.

## Task

- [ ] Link to the kibana dashboard with the insights of the build

## Related issues
Closes #ISSUE

### CODE REVIEW

File: resources/github-comment-markdown.template [hunk 1]
```diff
@@ -15,15 +15,15 @@
...
+[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](${jobUrl}/pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](${jobUrl}/tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](${jobUrl}/changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](${jobUrl}/artifacts) <% if(docsUrl?.trim() && !buildStatus?.equals('ABORTED')) {%>[![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](${docsUrl})<%}%>  <% if(env?.OTEL_ELASTIC_URL != null && !env?.OTEL_ELASTIC_URL.equals('null')) {%>[![preview](https://img.shields.io/badge/elastic-observability-blue)](${env?.OTEL_ELASTIC_URL})<%}%>
 
 <!-- BUILD SUMMARY-->
 <details><summary>Expand to view the summary</summary>
 <p>
 
 #### Build stats
 
-<% if(build?.causes?.shortDescription != null && !build?.causes?.shortDescription.equals('null')) {%>
+<% if(!build?.causes instanceof net.sf.json.JSONNull && build?.causes?.shortDescription != null && ! build?.causes?.shortDescription?.isEmpty() && !build?.causes?.shortDescription.equals('null')) {%>
```
Comment (v1v) [inline resources/github-comment-markdown.template:26]: If empty BO responses then

image

File: resources/groovy-html-custom.template [hunk 2]
```diff
@@ -135,7 +135,9 @@
       </tr>
       <tr>
         <td colspan="7" style="padding-left: 24px;text-align: left; vertical-align: middle; font-family: Helvetica; font-size: 12px; color: #343B49; letter-spacing: 1px;">
+          <% if(!build?.causes instanceof net.sf.json.JSONNull) {%>
```
Comment (v1v) [inline resources/groovy-html-custom.template:138]: See https://github.com/elastic/apm-pipeline-library/pull/1020/files#r589537029

File: vars/notifyBuildResult.groovy [hunk 3]
```diff
@@ -69,6 +69,8 @@ def call(Map args = [:]) {
       def aggregateComments = args.get('aggregateComments', true)
       def flakyDisableGHIssueCreation = args.get('flakyDisableGHIssueCreation', false)
       catchError(message: 'There were some failures with the notifications', buildResult: 'SUCCESS', stageResult: 'UNSTABLE') {
+        // FIXME: EnvironmentContributions are not populated in the env global map
+        setEnvVar('OTEL_ELASTIC_URL', sh(label: 'OTEL_ELASTIC_URL', script: 'echo -n "${OTEL_ELASTIC_URL}"', returnStdout: true))
```
Comment (v1v) [inline vars/notifyBuildResult.groovy:73]: EnvironmentContributions are only available in the shell context rather than the global `env` variable.

Comment (v1v) [reply vars/notifyBuildResult.groovy:73]: As soon as https://github.com/jenkinsci/opentelemetry-plugin/pull/71 gets merged and released then we won't need this fix

Comment (v1v) [reply vars/notifyBuildResult.groovy:72]:
```suggestion
```

**Review by cachedout — APPROVED** [review]

**Review by kuisathaverat — APPROVED** [review]

**Pull request merged**